src/rpc_mock.rs

```diff
@@ -0,0 +1,155 @@
+// Implementation of RpcRequestHandler trait for testing Rpc requests without i/o
+
+use crate::rpc_request::{RpcClient, RpcError, RpcRequestHandler};
+use serde_json::{self, Number, Value};
+use solana_sdk::hash::Hash;
+use solana_sdk::pubkey::Pubkey;
+use solana_sdk::signature::{Keypair, KeypairUtil};
+use solana_sdk::system_transaction::SystemTransaction;
+use solana_sdk::transaction::Transaction;
+use std::error;
+use std::io::{Error, ErrorKind};
+use std::net::SocketAddr;
+
+pub enum MockRpcRequest {
+    ConfirmTransaction,
+    GetAccountInfo,
+    GetBalance,
+    GetConfirmationTime,
+    GetLastId,
+    GetSignatureStatus,
+    GetTransactionCount,
+    RequestAirdrop,
+    SendTransaction,
+    RegisterNode,
+    SignVote,
+    DeregisterNode,
+    GetStorageMiningLastId,
+    GetStorageMiningEntryHeight,
+    GetStoragePubkeysForEntryHeight,
+}
+
+impl RpcRequestHandler for MockRpcRequest {
+    fn make_rpc_request(
+        &self,
+        client: &RpcClient,
+        _id: u64,
+        params: Option<Value>,
+    ) -> Result<Value, Box<dyn error::Error>> {
+        match &client.addr as &str {
+            "succeeds" => {
+                match self {
+                    MockRpcRequest::ConfirmTransaction => {
+                        if let Some(Value::Array(param_array)) = params {
+                            if let Value::String(param_string) = &param_array[0] {
+                                if param_string == "43yNSFC6fYTuPgTNFFhF4axw7AfWxB2BPdurme8yrsWEYwm8299xh8n6TAHjGymiSub1XtyxTNyd9GBfY2hxoBw8" {
+                                    Ok(Value::Bool(true))
+                                } else {
+                                    Ok(Value::Bool(false))
+                                }
+                            } else {
+                                Err(RpcError::RpcRequestError("Missing parameter".to_string()))?
```

Comment:
Looks like you could flatten this code quite a bit by doing a few early returns and then moving the `Ok()` wrapper to the last line of this function.